types/slate-react/index.d.ts

```diff
@@ -99,7 +99,8 @@ export interface RenderInlineProps extends RenderNodeProps {
     node: Inline;
 }
 
-export type EventHook = (event: Event, editor: CoreEditor, next: () => any) => any;
+export type EventHookT<T> = (event: T, editor: CoreEditor, next: () => any) => any;
+export type EventHook = EventHookT<Event>;
```

Comment:
Could avoid defining two types by just having
```ts
export type EventHook<T = Event> = (event: T, editor: CoreEditor, next: () => any) => any;
```

Reply:
Oops. I'm not familiar with the tooling and was trying to make it work with TS 1.x.
Updated.